src/Bicep.LangServer/Program.cs

```diff
@@ -9,12 +9,38 @@ namespace Bicep.LanguageServer
     public class Program
     {
         public static async Task Main(string[] args)
+            => await RunWithCancellation(async cancellationToken =>
+            {
+                // the server uses JSON-RPC over stdin & stdout to communicate,
+                // so be careful not to use console for logging!
+                var server = new Server(Console.OpenStandardInput(), Console.OpenStandardOutput());
+
+                await server.Run(cancellationToken);
+            });
+
+        private static async Task RunWithCancellation(Func<CancellationToken, Task> runFunc)
         {
-            // the server uses JSON-RPC over stdin & stdout to communicate,
-            // so be careful not to use console for logging!
-            var server = new Server(Console.OpenStandardInput(), Console.OpenStandardOutput());
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            Console.CancelKeyPress += (sender, e) =>
+            {
+                cancellationTokenSource.Cancel();
+                e.Cancel = true;
+            };
```

Comment:
Can you add a test for this?

Reply:
Tried this, couldn't get it working:
```
    [TestClass]
    public class ProgramTests
    {
        private static readonly byte[] ControlCByteSequence = new byte[] { 0x03 };

        [TestMethod]
        public async Task Program_terminates_cleanly_on_ctrl_c()
        {
            using var inputStream = new MemoryStream();
            using var inputReader = new StreamReader(inputStream);
            Console.SetIn(inputReader);

            var mainMethod = Program.Main(Array.Empty<string>());

            await inputStream.WriteAsync(ControlCByteSequence);
            await inputStream.FlushAsync();

            await mainMethod;
        }
    }
```

Reply:
It should be ok.

Reply:
I mean we can merge without the test for sure.